test/prefer-export-from.mjs

```diff
@@ -0,0 +1,287 @@
+			export const variable = namespace;
+		`,
+		outdent`
+			import * as namespace from 'foo';
+			namespace.bar = 1;
+			export {namespace as named};
+			export {namespace as default};
+			export const variable = namespace;
+		`,
+		// Some not exported
+		outdent`
+			import {named1, named2} from 'foo';
+			export {named1};
+		`,
+		outdent`
+			import defaultExport, {named} from 'foo';
+			export {defaultExport};
+		`,
+		outdent`
+			import defaultExport, {named} from 'foo';
+			export {named};
+		`,
+		outdent`
+			import defaultExport, * as namespace from 'foo';
+			export {defaultExport};
+		`,
+		// Existing export
+		outdent`
+			import * as foo from 'foo';
+			export {foo};
+			export * as bar from 'foo';
+		`,
+		outdent`
+			import * as foo from 'foo';
+			export {foo};
+			export {bar} from 'foo';
+		`,
+		outdent`
+			import * as foo from 'foo';
+			export {foo};
+			export {} from 'foo';
+		`,
+		outdent`
+			import * as foo from 'foo';
+			export {foo};
+			export * from 'foo';
+		`,
+		outdent`
+			import foo from 'foo';
+			export {foo};
+			export * as bar from 'foo';
+		`,
+		outdent`
+			import foo from 'foo';
+			export {foo};
+			export {bar} from 'foo';
+		`,
+		outdent`
+			import foo from 'foo';
+			export {foo};
+			export {bar,} from 'foo';
+		`,
+		outdent`
+			import foo from 'foo';
+			export {foo};
+			export {} from 'foo';
+		`,
+		outdent`
+			import foo from 'foo';
+			export {foo};
+			export * from 'foo';
+		`,
+		// Multiple
+		outdent`
+			import {named1, named2} from 'foo';
+			export {named1, named2};
+		`,
+		outdent`
+			import {named} from 'foo';
+			export {named as default, named};
+		`,
+		outdent`
+			import {named, named as renamed} from 'foo';
+			export {named, renamed};
+		`,
+		outdent`
+			import defaultExport, {named1, named2} from 'foo';
+			export {named1 as default};
+			export {named2};
+			export {defaultExport};
+		`,
+		outdent`
+			import * as foo from 'foo';
+			import * as bar from 'foo';
+			export {foo, bar};
+		`,
+		outdent`
+			import * as foo from 'foo';
+			export {foo, foo as bar};
+		`,
+		outdent`
+			import defaultExport from 'foo';
+			export * from 'foo';
+			export default defaultExport;
+		`,
+		outdent`
+			import defaultExport from 'foo';
+			export {named} from 'foo';
+			export * from 'foo';
+			export default defaultExport;
+		`,
+		outdent`
+			import defaultExport from './foo.js';
+			export {named} from './foo.js';
+			export default defaultExport;
+		`,
+		outdent`
+			import defaultExport from './foo.js';
+			export {named} from './foo.js?query';
+			export default defaultExport;
+		`,
+	],
+});
+
+// Strange case
+// TODO: confirm how should this work
+test.snapshot({
+	valid: [
+		outdent`
+			import * as namespace from 'foo';
+			export default namespace;
+			export {namespace};
+		`,
+	],
+	invalid: [
+		outdent`
+			import * as namespace from 'foo';
+			export {namespace};
+			export default namespace;
+		`,
+	],
+});
```

Comment:
These seem strange, I don't have time to confirm how should them work, but I guess we are fine to ignore it for now.

Reply:
Seems like a bug in ESLint. Both of those should be valid.